update 2.5

kndnsow opened the pull request.

Commit:
jquery & jqueryUI call from amd

Commit:
compleat js to amd conversion

kndnsow committed:
Update showhide.php

Commit:
amd external sortable

Commit:
amd ,externallib fix

amd with sortable and externallib with param db  compatability

purnendudash commented on externallib.php:
```diff
@@ -0,0 +1,81 @@
+            )
+        );
+        $order = $DB->get_record('block_myenrolledcoursesorder', array('userid' => $USER->id));
+        $neworder = new stdClass();
```
Why this

purnendudash commented on externallib.php:
```diff
@@ -0,0 +1,81 @@
+            $neworder->id = $order->id;
+            $DB->update_record('block_myenrolledcoursesorder', $neworder);
+        }
+        echo true;
```
Why this echo?

purnendudash commented on amd/src/myenrolledcourses.js:
```diff
@@ -0,0 +1,54 @@
+        }
+    });
+    },
+    disablebutten: function(){
```
Please use a proper function name

Commit:
final clean up for 2.5

Commit:
simplyfy

simplify the array to string and string to array to jeson.encode previously.
now its just string from start to end
